crates/matrix-sdk-ui/src/timeline/tests/mod.rs

```diff
@@ -165,31 +154,30 @@ impl TestTimeline {
     ) where
         C: RedactedStateEventContent,
     {
-        let ev = self.make_redacted_state_event(sender, state_key.as_ref(), content);
+        let ev =
+            self.event_builder.make_sync_redacted_state_event(sender, state_key.as_ref(), content);
         self.handle_live_event(Raw::new(&ev).unwrap().cast()).await;
     }
 
-    async fn handle_live_custom_event(&self, event: JsonValue) {
-        let raw = Raw::new(&event).unwrap().cast();
-        self.handle_live_event(raw).await;
+    async fn handle_live_custom_event(&self, event: Raw<AnySyncTimelineEvent>) {
+        self.handle_live_event(event).await;
     }
 
     async fn handle_live_redaction(&self, sender: &UserId, redacts: &EventId) {
-        let ev = json!({
+        let ev = sync_timeline_event!({
             "type": "m.room.redaction",
             "content": {},
             "redacts": redacts,
             "event_id": EventId::new(server_name!("dummy.server")),
             "sender": sender,
-            "origin_server_ts": self.next_server_ts(),
+            "origin_server_ts": self.event_builder.next_server_ts(),
```

Comment:
This is a bit of an abstraction leak here; does the `EventBuilder` provide a method to create such an event too, and could it have such a method if not?